src/buza/views.py

```diff
@@ -118,6 +118,30 @@ class SubjectDetail(generic.DetailView):
 class SubjectList(generic.ListView):
     model = models.Subject
 
+    # Order subjects that the user is following first, then by title.
+    ordering = ['-user_following', 'title']
+
+    def get_queryset(self) -> QuerySet:
+        """
+        Annotate each `Subject` with a `user_following` field,
+        which is true if the authenticated user is following it.
+        """
+        queryset: QuerySet = super().get_queryset()
+        user: models.RequestUser = self.request.user
+
+        if user.is_authenticated:
+            # Subquery existence check for relation to user:
+            return queryset.annotate(
+                user_following=Exists(
+                    models.User.objects.filter(pk=user.pk, subjects=OuterRef('pk')),
```

Comment:
`OuterRef` refers to a field in the queryset from which the annotate function is called, because filter is called for the User model.

Reply:
Yup. `OuterRef` is a slightly special part of Django's subquery support ([docs](https://docs.djangoproject.com/en/2.1/ref/models/expressions/#referencing-columns-from-the-outer-queryset)), in order to be able to express the reference to the outer query's column in Python.